src/classes/ALLO_ManageAllocations_CTRL.cls

```diff
@@ -150,132 +155,96 @@ public with sharing class ALLO_ManageAllocations_CTRL {
+        npe03__Recurring_Donation__c.SObjectType => System.Label.alloManageRecurringDonationAllocations,
+        npe01__OppPayment__c.SObjectType => System.Label.alloManagePaymentAllocations
+    };
 
+    /**
+     * @description Sets values for various allocation related member variables and adds a
+     * default GAU and/or a default allocation if necessary.
+     */
+    private void setupAllocations() {
         defaultAllo = new Allocation__c();
-
         listAllo = getAllocations(parentId);
 
-        if (settings.Default_Allocations_Enabled__c && (opp!=null || payment!=null)) {
-            for (integer i=0; i<listAllo.size(); i++) {
-                if (listAllo[i].General_Accounting_Unit__c == settings.Default__c) {
-                    defaultAllo = listAllo[i];
-                    listAllo.remove(i);
-                    break;
-                }
-            }
+        Boolean shouldSetDefaultGAU = settings.Default_Allocations_Enabled__c && (parent.hasAnAmountField());
+        if (shouldSetDefaultGAU) {
+            setDefaultGAU();
         }
 
         if (listAllo.isEmpty()) {
             addRow();
         }
-
-        isLoading = false;
     }
 
     /**
-     * @description Private default constructor
+     * @description Finds a matching default GAU and updates relevant member variables as needed.
      */
-    @TestVisible
-    private ALLO_ManageAllocations_CTRL() {}
-
-    /** @description Adds an empty row to the bottom of the list and refreshes the page.*/
-    private void populateObjectParameters(Schema.sObjectType st) {
-        DescribeSObjectResult describe = st.getDescribe();
-        keyPrefix = describe.getKeyPrefix();
-        objectAPIName = describe.getName();
-        objectLabelPlural = describe.getLabelPlural();
+    private void setDefaultGAU() {
+        for (integer i=0; i<listAllo.size(); i++) {
+            Boolean isDefaultGAU = listAllo[i].General_Accounting_Unit__c == settings.Default__c;
+            if (isDefaultGAU) {
+                defaultAllo = listAllo[i];
+                listAllo.remove(i);
+                break;
+            }
+        }
     }
 
     /*******************************************************************************************************
     * @description Verify data conditions and add any necessary page messages that need to be rendered on load.
     */
-    public void getPageLoadValidationMessages() {
-        if (parentAmount == 0 && sObjType == Opportunity.sObjectType) {
-            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.WARNING, System.Label.alloCantAllocateNothing));
-        } else if (!getIsSupportedObject()) {
+    public void verifyPageLoadConditions() {
+        if (!getIsSupportedObject()) {
             ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.WARNING, System.Label.alloObjectUnsupported));
+        } else if ((parent.isAnOpportunity() || parent.isAPayment()) && parent.getAmount() == 0) {
```

Comment:
Could this be rewritten as a function on the parent object? `parent.isEmptyAllocation()` or `parent.isInvalidAmount()`?